File: gcd/csharp_gcd/gcd.cs
using System;


public class GCDExample {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.Error.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit( 1 );
    }

    Console.WriteLine("gcd(6,4)=" + gcd(6,4));
    Console.WriteLine("gcd(45,5)=" + gcd(45,5));
    Console.WriteLine("gcd(45,7)=" + gcd(45,7));
  }

  public static int gcd( int a, int b ) {
    if( b == 0 ) {
      return a;
    } else {
      return gcd(b, (a % b));
    }
  }
}

File: gcd/fsharp_gcd/gcd.fs
#light
open System

//comment
(* commment *)

let rec gcd x y =
    if y = 0 then x
    else gcd y (x % y)

[<EntryPoint>]
let main args =
  let argv = Environment.GetCommandLineArgs()

  if argv.Length <> 1 then
    eprintfn "Usage: %s <noargs>" argv.[0]
    -1
  else
    printfn "(gcd 111 259)=%A" (gcd 111 259)
    printfn "(gcd 12 25)=%A" (gcd 12 25)
    printfn "(gcd 6 4)=%A" (gcd 6 4)
    printfn "(gcd 45 5)=%A" (gcd 45 5)
    0